src/cmd/faktory.go

```diff
@@ -130,24 +136,56 @@ func prepareJob(helper worker.Helper, job opslevel.RunnerJob) (opslevel.RunnerJo
 		Value:     string(job.Id),
 		Sensitive: false,
 	})
-	return job, nil
+	return nil
 }
 
-func parseJob(args []any) (opslevel.RunnerJob, error) {
-	data, err := json.Marshal(args[0])
-	if err != nil {
-		log.Error().Err(err).Msgf("failed to marshal job data: %v", args[0])
-		return opslevel.RunnerJob{}, err
+func extractCustomExtraFiles(helper worker.Helper, job *opslevel.RunnerJob) error {
+	extraFiles, ok := helper.Custom("opslevel-runner-files")
+	if ok {
+		var files []MapStructureRunnerJobFile
+		err := mapstructure.Decode(extraFiles, &files)
+		if err != nil {
+			return err
+		}
+		for _, file := range files {
+			job.Files = append(job.Files, opslevel.RunnerJobFile{
+				Name:     file.Name,
+				Contents: file.Contents,
+			})
+		}
 	}
-	var job opslevel.RunnerJob
-	if err = json.Unmarshal(data, &job); err != nil {
-		log.Error().Err(err).Msgf("failed to unmarshal job data: %v", string(data))
-		return opslevel.RunnerJob{}, err
+	return nil
+}
+
+func prepareJob(helper worker.Helper, args ...interface{}) (*opslevel.RunnerJob, error) {
+	var job *opslevel.RunnerJob
+
+	if err := parseJob(args, job); err != nil {
+		return nil, err
 	}
+
+	extractJobId(helper, job)
+
+	if err := extractCustomImage(helper, job); err != nil {
+		return nil, err
+	}
+
+	if err := extractCustomExtraCommands(helper, job); err != nil {
+		return nil, err
+	}
+
+	if err := extractCustomExtraVars(helper, job); err != nil {
+		return nil, err
+	}
+
+	if err := extractCustomExtraFiles(helper, job); err != nil {
+		return nil, err
+	}
+
```

Comment:
This is really the meat of the change - i broke all the different things out into their own functions to effectively prepare the job by extracting the necessary data if its available.  If any of these steps fail return an error so that we don't actually "run" the job